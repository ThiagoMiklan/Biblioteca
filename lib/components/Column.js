"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var React = _interopRequireWildcard(require("react"));

var _type_validations = require("../tools/type_validations.js");

var props_obj = {
  definition: "string"
};

var Column = function Column(props) {
  (0, _type_validations.validate)(props, props_obj, "Column");
  var definition = props.definition == undefined ? "column" : "column " + props.definition;
  return /*#__PURE__*/React.createElement("div", {
    className: definition
  }, props.children);
};

var _default = Column;
exports.default = _default;