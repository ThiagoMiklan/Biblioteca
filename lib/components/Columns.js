"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var React = _interopRequireWildcard(require("react"));

var _type_validations = require("../tools/type_validations.js");

var props_obj = {
  definition: 'string'
};

var Columns = function Columns(props) {
  (0, _type_validations.validate)(props, props_obj, "Columns");
  var definition = props.definition == undefined ? "columns" : "columns " + props.definition;
  return /*#__PURE__*/React.createElement("div", {
    className: definition
  }, React.Children.map(props.children, function (child, i) {
    return child;
  }));
};

var _default = Columns;
exports["default"] = _default;