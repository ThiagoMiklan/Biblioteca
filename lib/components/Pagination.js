"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _typeof2 = _interopRequireDefault(require("@babel/runtime/helpers/typeof"));

var _react = _interopRequireDefault(require("react"));

var _type_validations = require("../tools/type_validations");

var props_obj = {
  definition: "string",
  href_previous: "string",
  href_next: "string",
  next_disabled: "boolean",
  next_name: "string",
  previous_disabled: "boolean",
  itens: "array",
  previous_name: "string"
};

var Pagination = function Pagination(props) {
  (0, _type_validations.validate)(props, props_obj, "Pagination");
  var definition = props.definition == undefined ? "pagination" : "pagination " + props.definition;
  return /*#__PURE__*/_react.default.createElement("nav", {
    className: definition
  }, /*#__PURE__*/_react.default.createElement("a", {
    className: "pagination-previous",
    href: props.href_previous,
    disabled: props.previous_disabled
  }, props.previous_name), /*#__PURE__*/_react.default.createElement("a", {
    className: "pagination-next",
    href: props.href_next,
    disabled: props.next_disabled
  }, props.next_name), /*#__PURE__*/_react.default.createElement("ul", {
    class: "pagination-list"
  }, assemble(props.itens)));
};

function assemble(itens) {
  var code_itens = "";

  if (Array.isArray(itens)) {
    code_itens = assembleItens(itens);
  } else if ((0, _typeof2.default)(itens) == 'object') {
    code_itens = assembleItem(itens);
  }

  return code_itens;
}

function assembleItens(itens) {
  return itens.map(function (item) {
    return assembleItem(item);
  });
}

function assembleItem(item) {
  var definition = item["definition"] == undefined ? "pagination-link" : "pagination-link " + item["definition"];

  var item_code = item_code = /*#__PURE__*/_react.default.createElement("li", null, /*#__PURE__*/_react.default.createElement("a", {
    onClick: item["onClick"],
    disabled: item["disabled"],
    href: item["href"],
    className: definition
  }, item["value"]));

  return item_code;
}

var _default = Pagination;
exports.default = _default;