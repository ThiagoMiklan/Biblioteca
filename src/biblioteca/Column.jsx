import React from 'react';
import classnames from 'classnames';
import getClassName from '../tools/getClassName';
import PropTypes from 'prop-types';
/**
 * Componente inspirado na classe column do bulma
 * Utilizado como filho do componente Columns
 * Possui filhos, pois funciona como um container para outros
 * elementos
 * Sem eventos
 */

const Column = (props) => {
    var definition = classnames(getClassName("column "+ props.definition, "Column"));
    return (
        <div className={definition}>{props.children}</div>
    );
}

Column.propTypes = {
    // caracter√≠sticas do bulma para a coluna
    definition : PropTypes.string
}

export default Column;