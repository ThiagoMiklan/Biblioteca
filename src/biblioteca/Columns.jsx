import React from 'react';
import getClassName from '../tools/getClassName';
import classnames from 'classnames';
import PropTypes from 'prop-types';

/**
 *  Funciona como um container para Column
 *  Baseado na classe Columns do Bulma
 *  Sem eventos
 */

const Columns = (props)=>{
    var definition = classnames(getClassName("columns "+ props.definition,"Columns"));

    return <div className={definition}>
        {
            React.Children.map(props.children, (child, i)=>{
                return child;
            })
        }
    </div>

}

Columns.propTypes = {
    // Caracter√≠sticas do Bulma
    definition: PropTypes.string
}



export default Columns