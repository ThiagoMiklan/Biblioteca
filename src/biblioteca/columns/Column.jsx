import React from 'react';
import classnames from 'classnames';
import getClassName from '../../tools/getClassName';
import PropTypes from 'prop-types';

const Column = (props) => {
    var definition = classnames(getClassName("column "+ props.definition, "Column"));
    return (
        <div className={definition}>{props.children}</div>
    );
}

Column.propTypes = {
    definition : PropTypes.string
}

export default Column;