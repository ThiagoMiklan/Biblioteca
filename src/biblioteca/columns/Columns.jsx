import React from 'react';
import getClassName from '../../tools/getClassName';
import classnames from 'classnames';


const Columns = (props)=>{
    var definition = classnames(getClassName("columns "+ props.definition,"Columns"));

    return <div className={definition}>
        {
            React.Children.map(props.children, (child, i)=>{
                return child;
            })
        }
    </div>

}

export default Columns