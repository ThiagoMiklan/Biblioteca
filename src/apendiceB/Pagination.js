
/*
import React from 'react';
import ReactDOM from 'react-dom';
import 'bulma/css/bulma.css'
import Pagination from './components/Pagination';
import Columns from './components/Columns';
import Column from './components/Column';

var onClickItem = () => {
  alert("Clicked");
}

var itens = [
  { value: 1, href: "https://pt-br.reactjs.org/docs/hooks-overview.html", definition: "is-current", onClick: onClickItem },
  { value: 2, href: "https://pt-br.reactjs.org/docs/hooks-effect.html"},
  { value: 3, href: "https://pt-br.reactjs.org/docs/hooks-effect.html"}
]

ReactDOM.render(
  <nav class="pagination" role="navigation" aria-label="pagination">
  <a class="pagination-previous" title="This is the first page" disabled>Previous</a>
  <a class="pagination-next">Next page</a>
  <ul class="pagination-list">
    <li>
      <a class="pagination-link is-current" aria-label="Page 1" aria-current="page">1</a>
    </li>
    <li>
      <a class="pagination-link" aria-label="Goto page 2">2</a>
    </li>
    <li>
      <a class="pagination-link" aria-label="Goto page 3">3</a>
    </li>
  </ul>
</nav>
  ,
  document.getElementById('root'));

*/

/*
import React from 'react';
import ReactDOM from 'react-dom';
import 'bulma/css/bulma.css'
import Pagination from './components/Pagination';
import Columns from './components/Columns';
import Column from './components/Column';

var onClickItem = () => {
  alert("Clicked");
}

var itens = [
  { value: 1, href: "https://pt-br.reactjs.org/docs/hooks-overview.html", definition: "is-current", onClick: onClickItem },
  { value: 2, href: "https://pt-br.reactjs.org/docs/hooks-effect.html"},
  { value: 3, href: "https://pt-br.reactjs.org/docs/hooks-effect.html"}
]

ReactDOM.render(
  <Columns>
    <Column definition="is-half">
      <div className="container is-fluid"> 
      <br></br>
      <Pagination itens={itens} next_name="Next" previous_name="Previous"></Pagination>
      <Pagination itens={itens} next_name="Next" previous_name="Previous" previous_disabled={true}></Pagination>
      <Pagination itens={itens} next_name="Next" previous_name="Previous" previous_disabled={true} next_disabled={true}></Pagination>
      <Pagination definition="is-center" itens={itens} next_name="Next" previous_name="Previous"></Pagination>
      <Pagination definition="is-right" itens={itens} next_name="Next" previous_name="Previous"></Pagination>
      <Pagination definition="is-right is-rounded" itens={itens} next_name="Next" previous_name="Previous"></Pagination>
      
      <Pagination definition="is-right is-small" itens={itens} next_name="Next" previous_name="Previous"></Pagination>
      <Pagination definition="is-right is-medium" itens={itens} next_name="Next" previous_name="Previous"></Pagination>
      <Pagination definition="is-right is-large" itens={itens} next_name="Next" previous_name="Previous"></Pagination>
      </div>
    </Column>

  </Columns>
  ,
  document.getElementById('root'));


*/