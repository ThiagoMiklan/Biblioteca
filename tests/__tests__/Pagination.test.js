import React from 'react';
import Pagination from '../../src/components/Pagination';
import renderer from 'react-test-renderer';
import Columns from '../../src/components/Columns';
import Column from '../../src/components/Column';

var onClickItem = () => {
  alert("Clicked");
}

var itens = [
  { value: 1, href: "https://pt-br.reactjs.org/docs/hooks-overview.html", definition: "is-current", onClick: onClickItem },
  { value: 2, href: "https://pt-br.reactjs.org/docs/hooks-effect.html"},
  { value: 3, href: "https://pt-br.reactjs.org/docs/hooks-effect.html"}
]

test('value correctly', () => {
  const componentWithValue = renderer
    .create( 
      <Columns>
      <Column definition="is-half">
        <div className="container is-fluid"> 
        <br></br>
        <Pagination itens={itens} next_name="Next" previous_name="Previous"></Pagination>
        <Pagination itens={itens} next_name="Next" previous_name="Previous" previous_disabled={true}></Pagination>
        <Pagination itens={itens} next_name="Next" previous_name="Previous" previous_disabled={true} next_disabled={true}></Pagination>
        <Pagination definition="is-center" itens={itens} next_name="Next" previous_name="Previous"></Pagination>
        <Pagination definition="is-right" itens={itens} next_name="Next" previous_name="Previous"></Pagination>
        <Pagination definition="is-right is-rounded" itens={itens} next_name="Next" previous_name="Previous"></Pagination>
        
        <Pagination definition="is-right is-small" itens={itens} next_name="Next" previous_name="Previous"></Pagination>
        <Pagination definition="is-right is-medium" itens={itens} next_name="Next" previous_name="Previous"></Pagination>
        <Pagination definition="is-right is-large" itens={itens} next_name="Next" previous_name="Previous"></Pagination>
        </div>
      </Column>
  
    </Columns>

    )
    .toJSON();
  // componente com valor
  expect(componentWithValue).toMatchSnapshot();



});





