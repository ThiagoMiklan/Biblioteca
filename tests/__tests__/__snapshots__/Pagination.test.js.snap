// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`value correctly 1`] = `
<div
  className="columns"
>
  <div
    className="column is-half"
  >
    <div
      className="container is-fluid"
    >
      <br />
      <nav
        className="pagination"
      >
        <a
          className="pagination-previous"
        >
          Previous
        </a>
        <a
          className="pagination-next"
        >
          Next
        </a>
        <ul
          class="pagination-list"
        >
          <li>
            <a
              className="pagination-link is-current"
              href="https://pt-br.reactjs.org/docs/hooks-overview.html"
              onClick={[Function]}
            >
              1
            </a>
          </li>
          <li>
            <a
              className="pagination-link"
              href="https://pt-br.reactjs.org/docs/hooks-effect.html"
            >
              2
            </a>
          </li>
          <li>
            <a
              className="pagination-link"
              href="https://pt-br.reactjs.org/docs/hooks-effect.html"
            >
              3
            </a>
          </li>
        </ul>
      </nav>
      <nav
        className="pagination"
      >
        <a
          className="pagination-previous"
          disabled={true}
        >
          Previous
        </a>
        <a
          className="pagination-next"
        >
          Next
        </a>
        <ul
          class="pagination-list"
        >
          <li>
            <a
              className="pagination-link is-current"
              href="https://pt-br.reactjs.org/docs/hooks-overview.html"
              onClick={[Function]}
            >
              1
            </a>
          </li>
          <li>
            <a
              className="pagination-link"
              href="https://pt-br.reactjs.org/docs/hooks-effect.html"
            >
              2
            </a>
          </li>
          <li>
            <a
              className="pagination-link"
              href="https://pt-br.reactjs.org/docs/hooks-effect.html"
            >
              3
            </a>
          </li>
        </ul>
      </nav>
      <nav
        className="pagination"
      >
        <a
          className="pagination-previous"
          disabled={true}
        >
          Previous
        </a>
        <a
          className="pagination-next"
          disabled={true}
        >
          Next
        </a>
        <ul
          class="pagination-list"
        >
          <li>
            <a
              className="pagination-link is-current"
              href="https://pt-br.reactjs.org/docs/hooks-overview.html"
              onClick={[Function]}
            >
              1
            </a>
          </li>
          <li>
            <a
              className="pagination-link"
              href="https://pt-br.reactjs.org/docs/hooks-effect.html"
            >
              2
            </a>
          </li>
          <li>
            <a
              className="pagination-link"
              href="https://pt-br.reactjs.org/docs/hooks-effect.html"
            >
              3
            </a>
          </li>
        </ul>
      </nav>
      <nav
        className="pagination is-center"
      >
        <a
          className="pagination-previous"
        >
          Previous
        </a>
        <a
          className="pagination-next"
        >
          Next
        </a>
        <ul
          class="pagination-list"
        >
          <li>
            <a
              className="pagination-link is-current"
              href="https://pt-br.reactjs.org/docs/hooks-overview.html"
              onClick={[Function]}
            >
              1
            </a>
          </li>
          <li>
            <a
              className="pagination-link"
              href="https://pt-br.reactjs.org/docs/hooks-effect.html"
            >
              2
            </a>
          </li>
          <li>
            <a
              className="pagination-link"
              href="https://pt-br.reactjs.org/docs/hooks-effect.html"
            >
              3
            </a>
          </li>
        </ul>
      </nav>
      <nav
        className="pagination is-right"
      >
        <a
          className="pagination-previous"
        >
          Previous
        </a>
        <a
          className="pagination-next"
        >
          Next
        </a>
        <ul
          class="pagination-list"
        >
          <li>
            <a
              className="pagination-link is-current"
              href="https://pt-br.reactjs.org/docs/hooks-overview.html"
              onClick={[Function]}
            >
              1
            </a>
          </li>
          <li>
            <a
              className="pagination-link"
              href="https://pt-br.reactjs.org/docs/hooks-effect.html"
            >
              2
            </a>
          </li>
          <li>
            <a
              className="pagination-link"
              href="https://pt-br.reactjs.org/docs/hooks-effect.html"
            >
              3
            </a>
          </li>
        </ul>
      </nav>
      <nav
        className="pagination is-right is-rounded"
      >
        <a
          className="pagination-previous"
        >
          Previous
        </a>
        <a
          className="pagination-next"
        >
          Next
        </a>
        <ul
          class="pagination-list"
        >
          <li>
            <a
              className="pagination-link is-current"
              href="https://pt-br.reactjs.org/docs/hooks-overview.html"
              onClick={[Function]}
            >
              1
            </a>
          </li>
          <li>
            <a
              className="pagination-link"
              href="https://pt-br.reactjs.org/docs/hooks-effect.html"
            >
              2
            </a>
          </li>
          <li>
            <a
              className="pagination-link"
              href="https://pt-br.reactjs.org/docs/hooks-effect.html"
            >
              3
            </a>
          </li>
        </ul>
      </nav>
      <nav
        className="pagination is-right is-small"
      >
        <a
          className="pagination-previous"
        >
          Previous
        </a>
        <a
          className="pagination-next"
        >
          Next
        </a>
        <ul
          class="pagination-list"
        >
          <li>
            <a
              className="pagination-link is-current"
              href="https://pt-br.reactjs.org/docs/hooks-overview.html"
              onClick={[Function]}
            >
              1
            </a>
          </li>
          <li>
            <a
              className="pagination-link"
              href="https://pt-br.reactjs.org/docs/hooks-effect.html"
            >
              2
            </a>
          </li>
          <li>
            <a
              className="pagination-link"
              href="https://pt-br.reactjs.org/docs/hooks-effect.html"
            >
              3
            </a>
          </li>
        </ul>
      </nav>
      <nav
        className="pagination is-right is-medium"
      >
        <a
          className="pagination-previous"
        >
          Previous
        </a>
        <a
          className="pagination-next"
        >
          Next
        </a>
        <ul
          class="pagination-list"
        >
          <li>
            <a
              className="pagination-link is-current"
              href="https://pt-br.reactjs.org/docs/hooks-overview.html"
              onClick={[Function]}
            >
              1
            </a>
          </li>
          <li>
            <a
              className="pagination-link"
              href="https://pt-br.reactjs.org/docs/hooks-effect.html"
            >
              2
            </a>
          </li>
          <li>
            <a
              className="pagination-link"
              href="https://pt-br.reactjs.org/docs/hooks-effect.html"
            >
              3
            </a>
          </li>
        </ul>
      </nav>
      <nav
        className="pagination is-right is-large"
      >
        <a
          className="pagination-previous"
        >
          Previous
        </a>
        <a
          className="pagination-next"
        >
          Next
        </a>
        <ul
          class="pagination-list"
        >
          <li>
            <a
              className="pagination-link is-current"
              href="https://pt-br.reactjs.org/docs/hooks-overview.html"
              onClick={[Function]}
            >
              1
            </a>
          </li>
          <li>
            <a
              className="pagination-link"
              href="https://pt-br.reactjs.org/docs/hooks-effect.html"
            >
              2
            </a>
          </li>
          <li>
            <a
              className="pagination-link"
              href="https://pt-br.reactjs.org/docs/hooks-effect.html"
            >
              3
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</div>
`;
